lib/trollop.rb

```diff
@@ -169,16 +169,22 @@ def opt name, desc="", opts={}
       when Date; :date
       when Array
         if opts[:default].empty?
-          raise ArgumentError, "multiple argument type cannot be deduced from an empty array for '#{opts[:default][0].class.name}'"
-        end
-        case opts[:default][0]    # the first element determines the types
-        when Integer; :ints
-        when Numeric; :floats
-        when String; :strings
-        when IO; :ios
-        when Date; :dates
+          if opts[:type]
+            raise ArgumentError, "multiple argument type must be plural" unless MULTI_ARG_TYPES.include?(opts[:type])
+            nil
```

Comment:
This section of code is setting the local variable type_from_default.  Since the default here is an empty array, we cannot derive it--nil is therefore what I "expect" the value to be.  (In fact, the nil is only there for clarity--if the unless falls through, the return is nil.)

Following the code, this shortcuts the test following immediately, and is a nop in the assignment following the test.